### PR TITLE
Fix scientific notation. 

### PR DESCRIPTION
**Pull request by scarabus**
RAP-4318 JacksonUtil.prettify is really inefficient

### CODE REVIEW

**Comment by jeremy-flieder** [comment]
I would like to try and test this locally. Pulling from Dave's github and testing

File: Libs/RaptureCommon/src/main/java/rapture/common/impl/jackson/JacksonUtil.java [hunk 1]
```diff
@@ -139,6 +162,13 @@ public static String jsonFromObject(Object object) {
         return object;
     }
 
+    /**
+     * Very inefficient.
```
Comment (jonathan-major) [inline Libs/RaptureCommon/src/main/java/rapture/common/impl/jackson/JacksonUtil.java]: is this a TODO?

File: Libs/Reflex/src/main/java/reflex/AbstractReflexDataHandler.java [hunk 2]
```diff
@@ -78,10 +79,10 @@ public ReflexValue pullData(RaptureURI uri) {
                 String contentType = blobContainer.getHeaders().get("Content-Type");
                 if (contentType != null) {
                     if (contentType.startsWith(CSV)) {
-                        List<ReflexValue> linesArray = new ArrayList<ReflexValue>();
+                        List<ReflexValue> linesArray = new ArrayList<>();
```
Comment (amkimian) [inline Libs/Reflex/src/main/java/reflex/AbstractReflexDataHandler.java]: Why did you change this?

Comment (amkimian) [inline Libs/Reflex/src/main/java/reflex/AbstractReflexDataHandler.java]: (And every other instance of that constructor)

File: Libs/RaptureCommon/src/main/java/rapture/common/impl/jackson/JacksonUtil.java [hunk 3]
```diff
@@ -86,22 +94,37 @@
...
+     * @param object
+     * @return
+     */
+    @Deprecated
     public static String jsonFromObject(Object object) {
-        StringWriter writer = new StringWriter();
+        return jsonFromObject(object, false);
+    }
+
+    public static String jsonFromObject(Object object, boolean pretty) {
```
Comment (amkimian) [inline Libs/RaptureCommon/src/main/java/rapture/common/impl/jackson/JacksonUtil.java]: Can you simply create a prettyfy named function and call this with that flag. When you use this function later on the boolean parameter and its effect is completely obscure...

i.e..

prettyJson(Object object) { return jsonFromObject(object, true); }

File: Libs/Reflex/src/main/java/reflex/AbstractReflexDataHandler.java (hunk 2 above)
Comment (scarabus) [inline Libs/Reflex/src/main/java/reflex/AbstractReflexDataHandler.java]: It was done automatically. 
I turned on the 'format on save' option in Eclipse. It corrects the formatting to the house style.
As a side-effect it does other things like adding any missing annotations (@Override) and applying the Diamond operator. There's probably a way to turn it off.

File: Libs/RaptureCommon/src/main/java/rapture/common/impl/jackson/JacksonUtil.java (hunk 1 above)
Comment (scarabus) [inline Libs/RaptureCommon/src/main/java/rapture/common/impl/jackson/JacksonUtil.java]: It's a comment for anyone who wants to know why the method is deprecated. 
AFAICT it's only ever used in conjunction with jsonFromObject so I added a flag and a better implementation and changed some of the calls (only a few unit tests still use it)

File: Libs/RaptureCommon/src/main/java/rapture/common/impl/jackson/JacksonUtil.java (hunk 3 above)
Comment (scarabus) [inline Libs/RaptureCommon/src/main/java/rapture/common/impl/jackson/JacksonUtil.java]: OK

**Comment by jeremy-flieder** [comment]
I tested this and it seems to be working; the small numbers are now written out in full to the blobs
